weave/trace/vals.py

```diff
@@ -643,7 +644,7 @@ def make_trace_obj(
             )
         )
         val = from_json(read_res.obj.val, val.entity + "/" + val.project, server)
-
+        initialize_object(val)
```

Comment:
The expected initialization path -- make sure the image is fully loaded before returning it